despatch_advice_import/wizard/despatch_advice_import.py

```diff
@@ -0,0 +1,327 @@
+            if (int(line["order_line_id"])) in lines_by_id:
+                lines_by_id[int(line["order_line_id"])]["qty"] += line["qty"]
+                lines_by_id[int(line["order_line_id"])]["backorder_qty"] += line[
+                    "backorder_qty"
+                ]
+                if "product_lot" in line:
+                    lines_by_id[int(line["order_line_id"])]["product_lot"].append(
+                        line["product_lot"]
+                    )
+                    lines_by_id[int(line["order_line_id"])]["product_lot"] = list(
+                        set(lines_by_id[int(line["order_line_id"])]["product_lot"])
+                    )
+                lines_by_id[int(line["order_line_id"])]["uom"]["unece_code"].append(
+                    line["uom"]["unece_code"]
+                )
+                lines_by_id[int(line["order_line_id"])]["uom"]["unece_code"] = list(
+                    set(lines_by_id[int(line["order_line_id"])]["uom"]["unece_code"])
+                )
+            else:
+                lines_by_id[int(line["order_line_id"])] = line
+                if "product_lot" in line:
+                    lines_by_id[int(line["order_line_id"])]["product_lot"] = [
+                        lines_by_id[int(line["order_line_id"])]["product_lot"]
+                    ]
+                lines_by_id[int(line["order_line_id"])]["uom"]["unece_code"] = [
+                    lines_by_id[int(line["order_line_id"])]["uom"]["unece_code"]
+                ]
+
+        lines = self.env["purchase.order.line"].browse(lines_by_id.keys())
+
+        for line in lines:
+            order = line.order_id
+            line_info = lines_by_id.get(line.id)
+
+            if line_info["ref"]:
+                if order.name != line_info["ref"]:
+                    bdio.user_error_wrap(
+                        "",
+                        "",
+                        _("No purchase order found for name %s.") % line_info["ref"],
+                    )
+            else:
+                if order.name != po_name:
+                    bdio.user_error_wrap(
+                        "", "", _("No purchase order found for name %s.") % po_name
+                    )
+            stock_moves = line.move_ids.filtered(
+                lambda x: x.state not in ("cancel", "done")
+            )
+            moves_qty = sum(stock_moves.mapped("product_qty"))
+            if line_info["qty"] == moves_qty:
+                self._process_accepted(stock_moves, parsed_order_document)
+            elif not line_info["qty"] and not line_info["backorder_qty"]:
+                self._process_rejected(stock_moves, parsed_order_document)
+            else:
+                self._process_conditional(stock_moves, parsed_order_document, line_info)
+
+    def _process_rejected(self, stock_moves, parsed_order_document):
+        parsed_order_document["chatter_msg"] = (
+            parsed_order_document["chatter_msg"] or []
+        )
```

Comment:
```suggestion
        parsed_order_document["chatter_msg"] = parsed_order_document.get("chatter_msg", [])
```
this should be done before `process_data` is called and removed from all the other places